### DISCUSSION TITLE
is com_intr() expected to work with intno == 0xe6

### DISCUSSION THREAD

**Discussion by andrewbird** [dosemu2/dosemu2#1751] [General]
I was thinking that it might be nice to switch the FOSSIL driver init program from assembler to C. I'm trying to make fossil.com a builtin, but whilst com_intr() can call int21 successfully, I don't see any evidence of a call in the log when the interrupt is 0xe6. Here's my work in progress.
https://github.com/andrewbird/dosemu2/tree/fossil05

**Comment by stsp** [comment]
Looks good to me.
What kind of evidence you don't
see? It should invoke dos_helper()
fn. You can run `dosemu -gdb`
and break to that fn?

**Reply by stsp** [reply]
Ah, you have a typo, you shift
subhelper by 16, 8 is enough.

**Reply by andrewbird** [reply]
> Ah, you have a typo, you shift
subhelper by 16, 8 is enough.

Quite right, thanks. I'll tidy it up and test it out.
